lib/internal/socket_list.js

```diff
@@ -95,14 +93,12 @@ function SocketListReceive(slave, key) {
 util.inherits(SocketListReceive, EventEmitter);
 
 SocketListReceive.prototype.add = function(obj) {
-  var self = this;
-
   this.connections++;
 
   // Notify previous owner of socket about its state change
-  obj.socket.once('close', function() {
-    self.connections--;
+  obj.socket.once('close', () => {
+    this.connections--;
 
-    if (self.connections === 0) self.emit('empty');
+    if (this.connections === 0) this.emit('empty', this);
```

Comment:
To send the context as a parameter seems ambigous. Since we are using arrow functions, why not just binding the method l.82 ? ``onempty.bind(this)`` ?